addons/website/static/src/snippets/s_rating/options.js

```diff
@@ -0,0 +1,184 @@
+                this.$target.attr('data-active-custom-icon', this.faClassActiveCustomIcons);
+                this.$target.attr('data-inactive-custom-icon', this.faClassInactiveCustomIcons);
+                this.$target.attr('data-icon', 'custom');
+                this.iconType = 'custom';
+                resolve();
+            });
+            dialog.on('closed', this, function () {
+                if (!this._saving) {
+                    resolve();
+                }
+            });
+            dialog.open();
+        });
+    },
+
+    /**
+     * Set the number of active icons.
+     *
+     * @see this.selectClass for parameters
+     */
+    activeIconsNumber: function (previewMode, widgetValue, params) {
+        if (widgetValue > 0) {
+            this.$target.find('.rating_icon').attr('data-icon-active', 'false');
+            this.$target.find('.rating_icon:nth-child(-n+' + widgetValue + ')').attr('data-icon-active', 'true');
+            this._renderIcons();
+        } else {
+            return;
+        }
+    },
+
+    /**
+     * Set the total number of icons.
+     *
+     * @see this.selectClass for parameters
+     */
+    totalIconsNumber: function (previewMode, widgetValue, params) {
+        let $icons = this.$target.find('.rating_icon');
+        const totalIcons = $($icons).length;
+        if (widgetValue < 1) {
+            return;
+        } else if (widgetValue > totalIcons) {
+            let i = 0;
+            const count = widgetValue - totalIcons;
+            for (i = 0; i < count; i++) {
+                $($icons[totalIcons - 1]).clone().insertAfter($($icons[totalIcons - 1])).before(' ');
+            }
+        } else if (widgetValue < totalIcons) {
+            this.$target.find(".rating_icon:nth-last-child(-n+" + (totalIcons - widgetValue) + ")").remove();
+        }
+    },
+
+    /**
+     * Set icon color.
+     *
+     * @see this.selectClass for parameters
+     */
+    setIconColor: function (previewMode, widgetValue, params) {
+        this.activeIconColor = this._ensureCssColor(this.$target.attr('data-active-icon-color'));
+        this.inactiveIconColor = this._ensureCssColor(this.$target.attr('data-inactive-icon-color'));
+        this._renderIcons();
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * render the icons according to whether it is active or inactive
+     *
+     * @private
+     * 
+     */
+    _renderIcons: function () {
+        const $icons = this.$target.find('.rating_icon');
+        let $activeIcons = $icons.filter('[data-icon-active="true"]');
+        let $inactiveIcons = $icons.filter('[data-icon-active="false"]');
+        const faInactive = (this.iconType === 'thumbs-up') ? 'thumbs-o-up' : this.iconType + '-o';
```

Comment:
Too magic, create an object which maps the active class to the inactive class and read that object. Otherwise it makes grepping the code harder and migration more difficult.
```javascript
const XXX =  {
    'fa-thumbs-up': 'fa-thumbds-o-up',
    ...
    ...
};
const faInactive = XXX[this.iconType];
```